clang/lib/Basic/Targets/AArch64.cpp

```diff
@@ -1046,6 +1082,29 @@ bool AArch64TargetInfo::handleTargetFeatures(std::vector<std::string> &Features,
       HasSVEB16B16 = true;
       HasSMEB16B16 = true;
     }
+
+    if (Feature == "+fp8")
+      HasFP8 = true;
+    if (Feature == "+fp8fma")
```

Comment:
`fp8...` feature extensions should also set `HasFP8`?

Reply:
I could. 
I am not sure I should reproduce the same we have in  lib/Target/AArch64/AArch64Features.td also here. 
Is that what you  are suggestion here?
At first glance it does not look like this is a representation of what we have in 
 lib/Target/AArch64/AArch64Features.td

Reply:
I don't fully understand the effect of setting these bools and how the dependencies in `AArch64Features.td` correspond to setting the feature macros.  What I'm mainly after is consistency and so for the SME2 extensions you're setting `HasSME2` as well.  So I guess the question is why that is necessary?  If it's not, then perhaps the way to go is to remove those rather than adding the ones I suggest.

The most important part to verify is that all the feature macros a user would expect to be defined by a specific +feat are in fact defined.  If that is managed by `AArch64Features.td` then great, if not then that would explain why the key feature dependencies are duplicated here.

Reply:
If I run -E I can see all the macros dependency defined without adding them here. So I think it is fine to only set the new feature.
Maybe we need to look to the other definitions and check if we need to clean them up too.